### ISSUE TITLE
Add pictures to quests

### ISSUE THREAD

**Issue by vidvidvid** [MetaFam/TheGame#1446]
### What would you like to be added?

- [ ] In the quest creation screen, please add a field for adding a picture to the quest.
- [ ] It would be great if editing of quests would include editing of their respective picture.

### Why is this needed?

It is needed in #1443 & consequently in the #1438 PR .

**Comment by dysbulic** [comment]
> **Is there a preferred method of storing the IPFS link?**

You were right with the `ipfs://` URL. You have to feed it to [`httpLink`](https://github.com/MetaFam/TheGame/blob/ec8182d7da906b702edecb116ebec75b728a509d/packages/utils/src/linkHelpers.ts#L7-L35) when displaying.

**Comment by thebeyondr** [comment]
> {...} You have to feed it to [`httpLink`](https://github.com/MetaFam/TheGame/blob/ec8182d7da906b702edecb116ebec75b728a509d/packages/utils/src/linkHelpers.ts#L7-L35) when displaying.

@dysbulic ok thanks, so you said displaying, but I'm storing... I think it would be useful to use `httpLink` before storing so retrievers just get the link when displaying.

Is that what you were implying or is there any reason thats not a good idea?

**Comment by dysbulic** [comment]
> I think it would be useful to use `httpLink` before storing so retrievers just get the link when displaying.

@thebeyondr, that would lock us into a particular gateway for delivering the IPFS content.

I actually want to switch the gateway from [`ipfs.io`](//ipfs.io) to [`w3s.link`](//w3s.link), which should be more performant, but it currently has a [mimetype issue](https://github.com/web3-storage/freeway/issues/19) with SVGs that causes them to be Cross-Origin Resource Blocked by Chrome.

Once that's resolved, I'll change the link template & all our images should see an improvement.

**Comment by thebeyondr** [comment]
Ah I see ... I'm wary of the lock in as well.. So where does that leave this issue?

Do I just save the CID as `image` or is it on hold till those fixes happen?

**Comment by peth-yursick** [comment]
Have you tested how fast it loads? Also, where are you uploading them @vidvidvid?

**Comment by vidvidvid** [comment]
@peth-yursick you mean in quest chains? we're using web3 storage & uploading through IPFS

**Comment by olizilla** [comment]
We've got a fix for https://github.com/web3-storage/freeway/issues/19 which I'll get released today.

**Comment by dysbulic** [comment]
> Do I just save the CID as `image` or is it on hold till those fixes happen?

@thebeyondr, save as `ipfs://`, use `httpLink` when displaying.

**Comment by thebeyondr** [comment]
@dysbulic gotcha, thanks!
